### PR TITLE
Fix - codelab folder structure, add NotFound component

### PR DESCRIPTION
**Pull request by Reactiver**

### CODE REVIEW

**Comment by Reactiver** [comment]
Add angular routes in separate component.
https://user-images.githubusercontent.com/38430641/64281744-79dd5500-cf5c-11e9-87d6-b6c52df7bbbe.png

File: apps/codelab/src/app/codelabs/not-found/not-found.component.scss [hunk 1]
```diff
@@ -0,0 +1,53 @@
+.container {
```
Comment (NothingEverHappens) [inline apps/codelab/src/app/codelabs/not-found/not-found.component.scss]: You could technically drop .container wrapper and use :host insted?

File: apps/codelab/src/app/codelabs/not-found/not-found.component.scss [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  align-items: center;
+  height: 100%;
+  font-family: 'Helvetica Neue', sans-serif;
+
+  .message {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    .left {
```
Comment (NothingEverHappens) [inline apps/codelab/src/app/codelabs/not-found/not-found.component.scss]: Would be nice to have more semantic names than left and right

File: apps/codelab/src/app/codelabs/not-found/not-found.component.scss [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      font-size: 70px;
+      margin-right: 20px;
+    }
+
+    .right {
+      .big-text {
```
Comment (NothingEverHappens) [inline apps/codelab/src/app/codelabs/not-found/not-found.component.scss]: Same as above, maybe .header / .sub-header?

File: apps/codelab/src/app/codelabs/not-found/not-found.component.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'codelab-not-found',
+  templateUrl: './not-found.component.html',
+  styleUrls: ['./not-found.component.scss']
```
Comment (NothingEverHappens) [inline apps/codelab/src/app/codelabs/not-found/not-found.component.ts]: Add `changeDetection: ChangeDetectionStrategy.OnPush`
While there's not much value in it in 404 component, it's nice to have it for consistency.

File: apps/codelab/src/app/components/angular-routes/angular-routes.component.html [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<div class="container">
+  <ol>
+    <ng-container *ngFor="let route of menuRoutes">
```
Comment (NothingEverHappens) [inline apps/codelab/src/app/components/angular-routes/angular-routes.component.html]: Drop thecontainer, move ngFor on li item?

File: apps/codelab/src/app/components/angular-routes/angular-routes.component.scss [hunk 6]
```diff
@@ -0,0 +1,10 @@
+.container {
```
Comment (NothingEverHappens) [inline apps/codelab/src/app/components/angular-routes/angular-routes.component.scss]: You can also drop container wrapper?

File: apps/codelab/src/app/components/angular-routes/angular-routes.component.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import { Component } from '@angular/core';
+import { menuRoutes } from '../../codelabs/angular/angular-routing.module';
+
+@Component({
+  selector: 'codelab-angular-routes',
+  templateUrl: 'angular-routes.component.html',
+  styleUrls: ['angular-routes.component.scss']
```
Comment (NothingEverHappens) [inline apps/codelab/src/app/components/angular-routes/angular-routes.component.ts]: OnPush

File: apps/codelab/src/app/components/angular-routes/angular-routes.component.ts [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+})
+export class AngularRoutesComponent {
+  public menuRoutes = menuRoutes;
+}
```
Comment (NothingEverHappens) [inline apps/codelab/src/app/components/angular-routes/angular-routes.component.ts]: Add missing new line?

File: apps/codelab/src/app/codelabs/codelabs-routing.module.ts [hunk 9]
```diff
@@ -19,11 +20,20 @@ const routes: Routes = [
...
-    loadChildren: () => import('../modules/admin/admin.module').then(m => m.AdminModule)
+    loadChildren: () => import('./admin/admin.module').then(m => m.AdminModule)
   },
   {
     path: 'sync',
-    loadChildren: () => import('../modules/sync/sync.module').then(m => m.SyncAdminModule)
+    loadChildren: () => import('./sync/sync.module').then(m => m.SyncAdminModule)
+  },
+  {
+    path: 'typescript',
```
Comment (alex-bu-93) [inline apps/codelab/src/app/codelabs/codelabs-routing.module.ts]: Could you move it to angular-routing module? Inside of angular-routing it should be the first item of Routes and redirect to 'typescript' not 'angular/typescript'

File: apps/codelab/src/app/codelabs/not-found/not-found.component.scss (hunk 1 above)
Comment (Reactiver) [reply apps/codelab/src/app/codelabs/not-found/not-found.component.scss]: Yes, it's good idea! Done

File: apps/codelab/src/app/codelabs/not-found/not-found.component.scss (hunk 2 above)
Comment (Reactiver) [reply apps/codelab/src/app/codelabs/not-found/not-found.component.scss]: Renamed

File: apps/codelab/src/app/codelabs/not-found/not-found.component.scss (hunk 3 above)
Comment (Reactiver) [reply apps/codelab/src/app/codelabs/not-found/not-found.component.scss]: Done

File: apps/codelab/src/app/codelabs/not-found/not-found.component.ts (hunk 4 above)
Comment (Reactiver) [reply apps/codelab/src/app/codelabs/not-found/not-found.component.ts]: Ok. Done

File: apps/codelab/src/app/components/angular-routes/angular-routes.component.html (hunk 5 above)
Comment (Reactiver) [reply apps/codelab/src/app/components/angular-routes/angular-routes.component.html]: Ok. Moved ngFor on li item

File: apps/codelab/src/app/components/angular-routes/angular-routes.component.scss (hunk 6 above)
Comment (Reactiver) [reply apps/codelab/src/app/components/angular-routes/angular-routes.component.scss]: Yes. Deleted container and  used  :host instead

File: apps/codelab/src/app/components/angular-routes/angular-routes.component.ts (hunk 7 above)
Comment (Reactiver) [reply apps/codelab/src/app/components/angular-routes/angular-routes.component.ts]: Done

File: apps/codelab/src/app/components/angular-routes/angular-routes.component.ts (hunk 8 above)
Comment (Reactiver) [reply apps/codelab/src/app/components/angular-routes/angular-routes.component.ts]: Added

**Comment by kirjs** [comment]
LGTM from my side
@Reactiver Looks like this branch has conflicts, please update
@alex-bu-93 Feel free to merge when looks good

**Pull request merged**